### PR TITLE
[VL] Follow-ups for #7304

### PR DESCRIPTION
**Pull request by zhztheplayer**
Follow-ups for #7304

To keep the new module dependency graph defined in https://github.com/apache/incubator-gluten/issues/7264.

cc @zml1206 @PHILO-HE

### CODE REVIEW

File: gluten-arrow/src/main/scala/org/apache/spark/sql/utils/SparkArrowUtil.scala [hunk 1]
```diff
@@ -97,7 +98,7 @@ object SparkArrowUtil {
             .map(
               field =>
                 toArrowField(
-                  ConverterUtils.normalizeStructFieldName(field.name),
+                  normalizeColName(field.name),
```
Comment (PHILO-HE) [inline gluten-arrow/src/main/scala/org/apache/spark/sql/utils/SparkArrowUtil.scala:101]: @zhztheplayer, looks `normalizeColName` is not equivalent to `normalizeStructFieldName` which makes the conversion disabled for CH backend.

Comment (zml1206) [reply gluten-arrow/src/main/scala/org/apache/spark/sql/utils/SparkArrowUtil.scala:101]: It seems gluten-arrow only used for velox backend.

Comment (PHILO-HE) [reply gluten-arrow/src/main/scala/org/apache/spark/sql/utils/SparkArrowUtil.scala:101]: I see. Thanks!

**Review by PHILO-HE — APPROVED** [review]

File: gluten-arrow/src/main/scala/org/apache/spark/sql/utils/SparkArrowUtil.scala (hunk 1 above)
Comment (zhztheplayer) [reply gluten-arrow/src/main/scala/org/apache/spark/sql/utils/SparkArrowUtil.scala:101]: Yes. Let's keep `gluten-arrow` an individual module so future refactors can be made easier. Minor code duplications like this one will be acceptable.

**Pull request merged**